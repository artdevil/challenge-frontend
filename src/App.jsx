import React, { Component } from 'react';
import { ApolloClient, ApolloProvider, createNetworkInterface } from 'react-apollo';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';
import AddToDo from './components/add_todo';
import Todos from './components/todos';
import ActionTodos from './components/action_todos';

import './App.css';

const networkInterface = createNetworkInterface({
  uri: 'http://localhost:8080',
});

const client = new ApolloClient({
  networkInterface,
  connectToDevTools: true,
});

class App extends Component {
  render() {
    return (
      <ApolloProvider client={client}>
        <NavigationDrawer toolbarTitle="KeetHealth Test" includeDrawerHeader={false} defaultVisible={false}>
          <div className="md-grid">
            <ActionTodos client={client} />
            <AddToDo />
            <Todos />
          </div>
        </NavigationDrawer>
      </ApolloProvider>
    );
  }
}

export default App;
