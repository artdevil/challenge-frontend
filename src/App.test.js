import React from 'react';
import { shallow } from 'enzyme';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import App from './App';
import ActionTodos from './components/action_todos';
import AddToDo from './components/add_todo';
import Todos from './components/todos';

it('shallow renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('renders one <NavigationDrawer /> components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(NavigationDrawer).length).toBe(1);
});

it('renders one <ActionTodos /> components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(ActionTodos).length).toBe(1);
});

it('renders one <Todos /> components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Todos).length).toBe(1);
});

it('renders one <AddToDo /> components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(AddToDo).length).toBe(1);
});
