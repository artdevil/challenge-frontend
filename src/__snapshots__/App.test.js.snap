// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders without crashing 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": [Function],
      "dataIdFromObject": [Function],
      "disableNetworkFetches": false,
      "fieldWithArgs": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "http://localhost:8080",
      },
      "query": [Function],
      "queryDeduplication": true,
      "reducerConfig": Object {
        "addTypename": true,
        "customResolvers": undefined,
        "dataIdFromObject": [Function],
        "fragmentMatcher": [Function],
      },
      "resetStore": [Function],
      "setStore": [Function],
      "ssrMode": false,
      "version": "1.9.3",
      "watchQuery": [Function],
    }
  }
>
  <NavigationDrawer
    autoclose={true}
    constantDrawerType={false}
    contentComponent="main"
    contentId="main-content"
    defaultMedia="mobile"
    defaultVisible={false}
    desktopDrawerType="full-height"
    desktopMinWidth={1025}
    drawerTransitionDuration={300}
    extractMini={true}
    includeDrawerHeader={false}
    jumpLabel="Jump to content"
    mobileDrawerType="temporary"
    mobileMinWidth={320}
    persistentIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        arrow_back
      </FontIcon>
    }
    position="left"
    tabletDrawerType="persistent"
    tabletMinWidth={768}
    temporaryIcon={
      <FontIcon
        iconClassName="material-icons"
      >
        menu
      </FontIcon>
    }
    toolbarThemeType="colored"
    toolbarTitle="KeetHealth Test"
    transitionEnterTimeout={300}
    transitionName="md-cross-fade"
  >
    <div
      className="md-grid"
    >
      <Apollo(Apollo(ActionTodos)) />
      <Apollo(AddTodo) />
      <Apollo(Apollo(Apollo(Todos))) />
    </div>
  </NavigationDrawer>
</ApolloProvider>
`;
